includes/class-sensei-usage-tracking.php

```diff
@@ -0,0 +1,321 @@
+		$learners = $user_query->get_results();
+
+		foreach( $learners as $learner ) {
+			$course_args = array(
+				'user_id' => $learner->ID,
+				'type' => 'sensei_course_status',
+				'status' => 'any',
+			);
+
+			$course_count = Sensei_Utils::sensei_check_for_activity( $course_args );
+
+			if ( $course_count > 0 ) {
+				$learner_count++;
+			}
+		}
+
+		return $learner_count;
+	}
+
+	/**
+	 *
+	 * Register usage submenu page.
+	 **/
+	function register_usage_submenu_page() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		add_submenu_page(
+			'sensei',
+			__( 'Usage Tracking', 'woothemes-sensei' ),
+			__( 'Usage Tracking', 'woothemes-sensei' ),
+			'manage_options',
+			'sensei_usage_tracking',
+			array( $this, 'render_usage_tracking_page' )
+		);
+	}
+
+	function render_usage_tracking_page() {
+		?>
+		<div class="wrap">
+
+			<div id="icon-woothemes-sensei" class="icon32"><br></div>
+			<h1><?php _e('Sensei Usage Tracking', 'woothemes-sensei'); ?></h1>
+			<form method="post" action="" name="update-sensei" class="upgrade">
+				<p>
+					<input id="update-sensei" type="hidden" value="send_data" name="usage_tracking_action">
+					<input id="send-data" class="button button-primary" type="submit" value="Send Data Now">
+					<?php wp_nonce_field( 'send_data_nonce' ); ?>
+				</p>
+			</form>
+		</div>
+		<?php
+	}
+
+	function add_setting_field( $fields ) {
+		// default-settings
+		$fields[ self::$usage_tracking_setting_name ] = array(
+			'name' => __( 'Enable usage tracking', 'woothemes-sensei' ),
+			'description' => __( 'Allow Sensei to anonymously track plugin usage**. No sensitive data is tracked.', 'woothemes-sensei' ),
+			'type' => 'checkbox',
+			'default' => false,
+			'section' => 'default-settings'
+		);
+
+		return $fields;
+	}
+
+	function maybe_display_tracking_opt_in() {
+		$opt_in_displayed = (bool) get_option( 'sensei_usage_tracking_opt_in_display' );
+		$user_tracking_enabled = (bool) get_option( self::$usage_tracking_setting_name );
+
+		if ( false && ! $user_tracking_enabled && ! $opt_in_displayed ) { ?>
+			<div class="notice">
+				<p><?php echo esc_html__( 'Help us make Sensei better! Allow Sensei to anonymously track plugin usage. No sensitive data is tracked.', 'woothemes-sensei' ); ?></p>
+				<form>
+					<input class="button button-primary" type="submit" value="Yes">
+				</form>
+				<form>
+					<input class="button" type="submit" value="No">
+				</form>
+			</div>
+		<?php
+		}
+	}
+
+	/**
+	 * Output Tracks script.
+	 */
+	function load_tracking_async_defer() {
+		$protocol = is_ssl() ? 'https://' : 'http://';
+	?>
+		<script type="text/javascript" src="<?php echo $protocol; ?>stats.wp.com/w.js?56" async defer></script>
+	<?php
+	}
+
+	function add_sensei_tracking_script() {
+		$user = wp_get_current_user();
+	?>
+		<script>
+			window.Sensei = window.Sensei || {};
+
+			Sensei.record_event = function( event_name, event_properties ) {
```

Comment:
Does this mean we are going to do site metrics tracking as well as actual user action tracking? If so maybe we should split these commits out to different branches and use different event prefixes because they are pretty different things so mixing them together could be confusing.

Reply:
Removed in https://github.com/Automattic/sensei/pull/1965/commits/3acd1e00d99efafc5260ae2b6ff51583fd3bdfbb.